Update Readme to redirect to the docs site (#277)
dbt-labs/dbt-mcp: README.md

![architecture diagram of the dbt MCP server](https://raw.githubusercontent.com/dbt-labs/dbt-mcp/refs/heads/main/docs/d2.png)

## Documentation

[To learn more about dbt MCP functionalities, setup and integrations, please go to our docs site.](https://docs.getdbt.com/docs/dbt-ai/about-mcp#mcp-integrations). 

> For integration guides, we do

Reviewer: I think you meant to finish this part?

![architecture diagram of the dbt MCP server](https://raw.githubusercontent.com/dbt-labs/dbt-mcp/refs/heads/main/docs/d2.png)

## Documentation

> [We have moved the documentation from the readme over to the dbt docs site. For setup information as well as functionalites, please go to our docs site.](https://docs.getdbt.com/docs/dbt-ai/about-mcp#mcp-integrations)

Reviewer: I don't think it is relevant to mention the history of the documentation here.

```suggestion
[For setup information and to learn more about dbt MCP functionalities, please go to our docs site.](https://docs.getdbt.com/docs/dbt-ai/about-mcp#mcp-integrations)
```

Author: So the reason I left this here is because it's possible some folks were coming back to the readme for the docs. I've seen it happen before where people were like am I crazy 😓   (especially if their browsers had cached anything)
